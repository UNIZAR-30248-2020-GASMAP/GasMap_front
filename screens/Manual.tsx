import * as React from 'react';
import { Image, StyleSheet } from 'react-native';

import { Text, View } from '../components/Themed';


export default class Manual extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.mainTitle}>User Manual</Text>
        <Text style={styles.title}>Main Screen</Text>
        <Text style={styles.plain}>The main screen shows a map where some gas
          stations are displayed with a petrol pump icon like this one.
          <Image source={require('../assets/images/gas-station-icon.png')}/>
        </Text>

        <Text style={styles.title}>Manager Login</Text>
        <Text style={styles.plain}>A manager need a valid account that will be
          validated through a form, wich requires an email and password.
          The email needs a valid format and the password has to be 12 characters
          minimum.
        </Text>

        <Text style={styles.title}>Gas station</Text>
        <Text style={styles.plain}>This screen shows all of the important information
        about the clicked gas station. It also allows the user to update the price of 
        one fuel type if it has not been changed in the last 24 hours and if it does not variate
        more than 5% of the last price. There is also a report button for the user to report
        anything that looks wrong.
        </Text>
        <Text style={styles.plain}>On this screen is also shown a graph with the last 5
        prices that a fuel type has.
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'baseline',
    backgroundColor: 'white'
  },
  mainTitle: {
    fontSize: 25,
    fontWeight: 'bold',
    paddingTop: '15%',
    paddingBottom: '10%',
    color: 'black',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    paddingTop: '5%',
    color: 'black',
  },
  plain: {
    fontSize: 20,
    color: 'black',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
});
