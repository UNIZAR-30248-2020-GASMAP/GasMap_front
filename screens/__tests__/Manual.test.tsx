import React from 'react';
import renderer from 'react-test-renderer';

import Manual from '../Manual';

test('Correct Rendered', () => {
    const tree = renderer.create(<Manual />).toJSON();
    expect(tree).toMatchSnapshot()
});