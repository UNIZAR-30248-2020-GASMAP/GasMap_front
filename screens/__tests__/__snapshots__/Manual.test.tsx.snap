// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct Rendered 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "baseline",
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "black",
          "fontSize": 25,
          "fontWeight": "bold",
          "paddingBottom": "10%",
          "paddingTop": "15%",
        },
      ]
    }
  >
    Manual de usuario
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingTop": "5%",
        },
      ]
    }
  >
    Vista Principal
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "black",
          "fontSize": 20,
        },
      ]
    }
  >
    The main screen shows a map where some gas stations are displayed with a petrol pump icon like this one.
    <Image
      source={1}
    />
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingTop": "5%",
        },
      ]
    }
  >
    Manager Login
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "black",
          "fontSize": 20,
        },
      ]
    }
  >
    A manager need a valid account that will be validated through a form, wich requires an email and password. The email needs a valid format and the password has to be 12 characters minimum.
  </Text>
</View>
`;
